### DISCUSSION TITLE
add nbtutor in the future (remember pythontutor.com)

### DISCUSSION THREAD

**Discussion by FrisoDB** [winpython/winpython#940] [Ideas]
When I started using python in 2014 we had to learn using the basic python idle. The big issue was of course debugging. At the time http://pythontutor.com/ was heaven on earth for many starting python programmers.

Nowadays everything has moved on to jupyter notebooks, which is greatly appreciated. Lately I found the package/project **nbtutor**, which has basicly the same functionality but integrates it into notebooks.
https://github.com/lgpage/nbtutor

My suggestion is to consider adding it to winpython when has been updated. Are there more people/teachers/students who think this package or similar things would be a good addition.

**Comment by stonebig** [comment]
only problem is that Jupyterlab seems not compatible with nbtutor, and now Logan Page wants to rewrite it in angular, after 4 years of silence. the rest is nice

**Reply by FrisoDB** [reply]
True. We have to see how that works out.

**Reply by FrisoDB** [reply]
https://github.com/alexmojaki/birdseye
Maybe this is something? At least it seems current.
